.github/workflows/rust-publish.yaml

```diff
@@ -0,0 +1,41 @@
+# This file is part of Astarte.
+#
+# Copyright 2023 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: Rust crate publish
+
+on:
+  create:
+
+jobs:
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: 1.59
+          override: true
+      - name: Publish astarte-message-hub-proto
+        run: cargo publish --token ${CRATES_TOKEN} --dry-run
+        env:
+          CRATES_TOKEN: ${{ secrets.CRATES_TOKEN }}
+        working-directory: ./rust/astarte-message-hub-proto
```

Comment:
**sudggestion**: add more checks for the rust code

We should add more check to the rust code to catch breakages in the generated code, I think at least the checks `clippy`, `doc` and `msrv` from the `astarte sdk`  [check.yaml](https://github.com/astarte-platform/astarte-device-sdk-rust/blob/master/.github/workflows/check.yaml) should be included.

Reply:
for the `msrv` the `-Z direct-minimal-versions` option in not present in the `1.59.1` toolchain so we skip this check, and introduce it when we will update `msrv` to `1.66.1`

Reply:
The minimal-version is different from the MSRV job

Reply:
Are you sure that `direct-minimal-versions` are supported in `1.59`?